app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewState.kt

```diff
@@ -8,8 +8,9 @@ data class ConversationViewState(
     val conversationName: String = "",
     val conversationAvatar: ConversationAvatar = ConversationAvatar.None,
     val messages: List<MessageViewWrapper> = emptyList(),
-    val onError: ConversationErrors? = null,
+    val onSnackbarMessage: ConversationSnackbarMessages? = null,
     val messageText: String = "",
+    val downloadedAssetDialogState: DownloadedAssetDialogVisibilityState = DownloadedAssetDialogVisibilityState.Hidden,
```

Comment:
honestly I do no think this is responsibility of the ViewModel, we could move the logic of showing and dismissing the dialog to ConversationScreenState, but I remember that dialogs will be refactored later on , so perhaps we can leave it as it is for now ?

Reply:
I think there is indeed room for improvement here. My main idea was to unify `ConversationScreenState` into `ConversationViewState` as they both hold visual information of the screen and it feels to me like an overkill having 2 states representing the same thing.

Reply:
I do not think ViewModel should have such details as showing dialog and stuff, ConversationScreenState is there to handle the UI related stuff. In my opinion ViewModel should expose just data and do not go as deep as managing the Dialog I think we discussed it during the Android Collective and there was general agreement on that